packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -152,31 +153,64 @@ export const makeZCFZygote = (
           zcfSeat = makeEmptySeatKit().zcfSeat;
         }
         const totalToMint = Object.values(gains).reduce(add, empty);
-        zcfSeat.incrementBy(gains);
+        assert(
+          !zcfSeat.hasExited(),
+          `zcfSeat must be active to mint gains for the zcfSeat`,
+        );
+        const allocationPlusGains = addToAllocation(
+          zcfSeat.getCurrentAllocation(),
+          gains,
+        );
+
+        // Increment the stagedAllocation if it exists so that the
+        // stagedAllocation is kept up to the currentAllocation
+        if (zcfSeat.hasStagedAllocation()) {
+          zcfSeat.incrementBy(gains);
+        }
```

Comment:
I see why this is necessary, but OMG what a mess. No change suggested.

Reply:
Hah, my thoughts as well.